### PR TITLE
Replaced unnecessary setUp() in the test suite.

### PR DESCRIPTION
**Pull request by charettes**
Whether or not it was safe to switch to `setUpTestData()` was investigated locally when working on a [per-test isolation feature of `setUpTestData()`](https://github.com/charettes/django-testdata) I'm planning to discuss on django-developers in a few hours.

### CODE REVIEW

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -3687,21 +3687,17 @@ class AdminInlineFileUploadTest(TestCase):
...
         filename = file1.name
         file1.close()
-        self.gallery = Gallery(name="Test Gallery")
-        self.gallery.save()
-        self.picture = Picture(name="Test Picture", image=filename, gallery=self.gallery)
-        self.picture.save()
+        cls.gallery = Gallery(name="Test Gallery")
+        cls.gallery.save()
+        cls.picture = Picture(name="Test Picture", image=filename, gallery=cls.gallery)
+        cls.picture.save()
```
Comment (felixxm) [inline tests/admin_views/tests.py]: Maybe `.objects.create(...)`? We will have two lines less :smile: Please change to single quotes.

File: tests/admin_views/tests.py [hunk 2]
```diff
@@ -3740,6 +3736,8 @@ class AdminInlineTests(TestCase):
     @classmethod
     def setUpTestData(cls):
         cls.superuser = User.objects.create_superuser(username='super', password='secret', email='super@example.com')
+        cls.collector = Collector(pk=1, name='John Fowles')
```
Comment (felixxm) [inline tests/admin_views/tests.py]: `.objects.create(...)`

File: tests/foreign_object/tests.py [hunk 3]
```diff
@@ -18,28 +18,28 @@
...
         # Creating People
-        self.bob = Person()
-        self.bob.name = 'Bob'
-        self.bob.person_country = self.usa
-        self.bob.save()
-        self.jim = Person.objects.create(name='Jim', person_country=self.usa)
-        self.george = Person.objects.create(name='George', person_country=self.usa)
+        cls.bob = Person()
+        cls.bob.name = 'Bob'
+        cls.bob.person_country = cls.usa
```
Comment (felixxm) [inline tests/foreign_object/tests.py]: We can move `name` and `person_country` to `Person(...)`.

File: tests/postgres_tests/test_array.py [hunk 4]
```diff
@@ -366,16 +367,17 @@ def test_unsupported_lookup(self):
...
         now = timezone.now()
-        self.datetimes = [now]
-        self.dates = [now.date()]
-        self.times = [now.time()]
-        self.objs = [
+        cls.datetimes = [now]
+        cls.dates = [now.date()]
+        cls.times = [now.time()]
+        cls.objs = [
             DateTimeArrayModel.objects.create(
```
Comment (felixxm) [inline tests/postgres_tests/test_array.py]: This can be single-lined.

File: tests/sites_tests/tests.py [hunk 5]
```diff
@@ -20,13 +20,14 @@
 class SitesFrameworkTests(TestCase):
     multi_db = True
 
-    def setUp(self):
-        self.site = Site(
+    @classmethod
+    def setUpTestData(cls):
+        cls.site = Site(
```
Comment (felixxm) [inline tests/sites_tests/tests.py]: This can be single-lined.

**Comment by charettes** [comment]
@felixxm I don't mind doing all of these changes in following commits but I wanted to keep the noise ratio as small as possible when doing the `setUp(self)` to `setUpTestData(cls)` switch.

**Comment by charettes** [comment]
@felixxm I just pushed 1c9af5ed1ceb60ab3ea0117ede9e1ae898dc8e5a to deal with your suggestions. I'll let fellows decide whether or not they want to squash it in 2d1d49d95ee08ea5f81e55755e6412971e548610.

**Comment by felixxm** [comment]
Thanks Simon :+1:

File: tests/auth_tests/test_decorators.py [hunk 6]
```diff
@@ -58,12 +58,15 @@ class PermissionsRequiredDecoratorTest(TestCase):
...
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = models.User.objects.create(username='joe', password='qwerty')
         # Add permissions auth.add_customuser and auth.change_customuser
         perms = models.Permission.objects.filter(codename__in=('add_customuser', 'change_customuser'))
-        self.user.user_permissions.add(*perms)
+        cls.user.user_permissions.add(*perms)
+
+    def setUp(self):
+        self.factory = RequestFactory()
```
[the remark this answers is not in the document]
Comment (charettes) [reply tests/auth_tests/test_decorators.py]: No it wasn't. I'll keep it in `setUp` in all instances.

**Comment by charettes** [comment]
@timgraham, I addressed your comments and pushed a following commit that defines reused `RequestFactory` instances as class attributes. That removes a few other unnecessary `setUp`.

**Pull request merged**